src/group_theory/finite_abelian.lean

```diff
@@ -17,15 +17,37 @@ import algebra.module.pid
 
 -/
 
-
 open_locale direct_sum
 
+universe u
+
+variables (G : Type u)
+
+namespace module
+
+lemma finite_of_fg_torsion [add_comm_group G] [module ℤ G] [module.finite ℤ G]
+  (hG : module.is_torsion ℤ G) : _root_.finite G :=
+begin
+  rcases module.equiv_direct_sum_of_is_torsion hG with ⟨ι, _, p, h, e, ⟨l⟩⟩,
+  haveI : ∀ i : ι, fact $ 0 < (p i ^ e i).nat_abs :=
+  λ i, fact.mk $ int.nat_abs_pos_of_ne_zero $ pow_ne_zero (e i) (h i).ne_zero,
+  haveI : ∀ i : ι, _root_.finite $ ℤ ⧸ submodule.span ℤ {p i ^ e i} :=
+  λ i, finite.of_equiv _ (p i ^ e i).quotient_span_equiv_zmod.symm.to_equiv,
+  haveI : _root_.finite ⨁ i, ℤ ⧸ (submodule.span ℤ {p i ^ e i} : submodule ℤ ℤ) :=
+  finite.of_equiv _ dfinsupp.equiv_fun_on_fintype.symm,
+  exact finite.of_equiv _ l.symm.to_equiv
+end
```

Comment:
Nitpick alert: Hmmz, can we rename the additive group/module to `A` or `M`? A `\Z`-module named `G` is a bit weird in my eyes.